### PR TITLE
new resource: `azurerm_private_dns_resolver_outbound_endpoint`

### PR DESCRIPTION
**Pull request by ms-henglu**
https://github.com/hashicorp/terraform-provider-azurerm/issues/16956

image

### CODE REVIEW

File: website/docs/r/private_dns_resolver_outbound_endpoint.html.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name which should be used for this Private DNS Resolver Outbound Endpoint. Changing this forces a new Private DNS Resolver Outbound Endpoint to be created.
+
+* `private_dns_resolver_private_dns_resolver_id` - (Required) Specifies the ID of the Private DNS Resolver Outbound Endpoint. Changing this forces a new Private DNS Resolver Outbound Endpoint to be created.
```
Comment (katbyte) [inline website/docs/r/private_dns_resolver_outbound_endpoint.html.markdown:46]: a copy paste error in this file?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
Hi @katbyte - Thanks for the code review. I've updated the docs and resolved the conflicts, please take another look, thanks!

File: internal/services/privatednsresolver/private_dns_resolver_outbound_endpoint_resource.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	networkValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type PrivateDNSResolverOutboundEndpointModel struct {
+	Name                            string            `tfschema:"name"`
+	PrivateDNSResolverDnsResolverId string            `tfschema:"private_dns_resolver_id"`
```
Comment (stephybun) [inline internal/services/privatednsresolver/private_dns_resolver_outbound_endpoint_resource.go:23]: Could this be 
```suggestion
	PrivateDnsResolverId string            `tfschema:"private_dns_resolver_id"`
```

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 💯

**Pull request merged**